nixos/modules/programs/firefox.nix

```diff
@@ -49,42 +51,106 @@ in {
         ${organisationInfo}
       '';
       type = policyFormat.type;
-      default = {};
+      default = { };
     };
 
     preferences = mkOption {
       description = mdDoc ''
-        Preferences to set from `about://config`.
+        Preferences to set from `about:config`.
 
         Some of these might be able to be configured more ergonomically
         using policies.
 
         ${organisationInfo}
       '';
       type = with types; attrsOf (oneOf [ bool int string ]);
-      default = {};
+      default = { };
+    };
+
+    preferencesStatus = mkOption {
+      description = mdDoc ''
+        The status of `firefox.preferences`.
+
+        `status` can be "default", "locked", "user" or "clear"
+        - `"default"`: Read/Write: Settings appear as default even if factory default differs.
+        - `"locked"`: Read-Only: Settings appear as default even if factory default differs.
+        - `"user"`: Read/Write: Settings appear as changed if it differs from factory default.
+        - `"clear"`: Read/Write: Value has no effect. Resets to factory defaults on each startup.
+      '';
+      type = types.enum [ "default" "locked" "user" "clear" ];
+      default = "locked";
+    };
+
+    autoConfig = mkOption {
+      description = mdDoc ''
+        AutoConfig files can be used to set and lock preferences that are not covered
+        by the policies.json for Mac and Linux. This method can be used to automatically
+        change user preferences or prevent the end user from modifiying specific
+        preferences by locking them.
+      '';
+      type = types.lines;
+      default = "";
+    };
+
+    nativeMessagingHosts = {
+      browserpass = mkEnableOption (mdDoc "Browserpass support");
+      bukubrow = mkEnableOption (mdDoc "Bukubrow support");
+      ff2mpv = mkEnableOption (mdDoc "ff2mpv support");
+      fxCast = mkEnableOption (mdDoc "fx_cast support");
+      gsconnect = mkEnableOption (mdDoc "GSConnect support");
+      jabref = mkEnableOption (mdDoc "JabRef support");
+      passff = mkEnableOption (mdDoc "PassFF support");
+      tridactyl = mkEnableOption (mdDoc "Tridactyl support");
+      ugetIntegrator = mkEnableOption (mdDoc "Uget Integrator support");
```

Comment:
Can we express this a bit smarter and shorter?

Reply:
I'm not sure... What's your suggestion?

Reply:
something like `mapAttrs (_: v: mkEnableOption (mdDoc v)) { browserpass = "Browserpass support"; … }` maybe?

Reply:
Ah, got it. Thanks!